### DISCUSSION TITLE
any plans to Direct solver for LCP like Roblox

### DISCUSSION THREAD

**Discussion by MyelinsheathXD** [google-deepmind/mujoco#265] [Ideas]
any plans to Direct solver for LCP like Roblox physics do,
Roblox team did a great job regarding to solve LCP  in real time  using direct method,
I think this feature gives great accuracy to robotics simulation

https://youtu.be/P-WP1yMOkc4

**Comment by yuvaltassa** [comment]
We have plans to implement [implicit complementarity](https://ieeexplore.ieee.org/document/5509739) at some point, but MuJoCo's current contact model is complementarity-free, for good reason. LCP's are non-convex and hard to solve. MuJoCo's Newton solver is by far the fastest and most accurate solver in any full featured simulator. To get a sense of how good it is, load any model and switch between Newton and PGS (what Roblox are using, though quite efficiently) while looking at the built-in profiler, and see how much better the Newton solver is.

**Reply by kevinzakka** [reply]
For a non-paywalled link: https://homes.cs.washington.edu/~todorov/papers/TodorovICRA10.pdf
